### PR TITLE
sync: various compatibility improvement

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

**Comment by sylvestre** [comment]
Passes:
`Congrats! The gnu test tests/misc/sync is no longer failing!`

File: src/uu/sync/src/sync.rs [hunk 1]
```diff
@@ -169,12 +175,45 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
...
-                1,
-                format!("cannot stat {}: No such file or directory", f.quote()),
-            ));
+        // Use the Nix open to be able to set the NONBLOCK flags for fifo files
+        #[cfg(any(target_os = "linux", target_os = "android"))]
+        {
+            match open(Path::new(&f), OFlag::O_NONBLOCK, Mode::empty()) {
+                Ok(_) => {}
+                Err(e) => {
+                    if e == Errno::ENOENT {
```
Comment (tertsdiepraam) [inline src/uu/sync/src/sync.rs:189]: We should probably add automatic conversion from `nix::Errno` to `UError` like we have for `std::io::Error`. Then we don't need these manual conversions.

Comment (sylvestre) [reply src/uu/sync/src/sync.rs:189]: Sounds like a good idea. Do you want me to open an issue as a good first bug ?

Comment (tertsdiepraam) [reply src/uu/sync/src/sync.rs:189]: Yeah good idea

**Comment by sylvestre** [comment]
@tertsdiepraam ok with you ? :)

**Comment by tertsdiepraam** [comment]
Yep, all good!

**Pull request merged**